sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpProducerTests.cs

```diff
@@ -408,6 +409,155 @@ public void SendEnumerableRespectsTheRetryPolicy(EventHubsRetryOptions retryOpti
                     ItExpr.IsAny<CancellationToken>());
         }
 
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpProducer.SendAsync" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCaseSource(nameof(RetryOptionTestCases))]
+        public void SendEnumerableConsidersOperationCanceledExceptionAsRetriable(EventHubsRetryOptions retryOptions)
+        {
+            var partitionId = "testMe";
+            var retriableException = new OperationCanceledException();
+            var retryPolicy = new BasicRetryPolicy(retryOptions);
+
+            var producer = new Mock<AmqpProducer>("aHub", partitionId, Mock.Of<AmqpConnectionScope>(), new AmqpMessageConverter(), retryPolicy)
+            {
+                CallBase = true
+            };
+
+            producer
+                .Protected()
+                .Setup<Task<SendingAmqpLink>>("CreateLinkAndEnsureProducerStateAsync",
+                    ItExpr.IsAny<string>(),
+                    ItExpr.IsAny<TimeSpan>(),
+                    ItExpr.IsAny<CancellationToken>())
+                 .Throws(retriableException);
```

Comment:
nit: `.Throws(...)` is not indented properly. It has one extra space. This file contains some more copy-pasted occurrences.